### DISCUSSION TITLE
invalid class "CellDataSet" object: sampleNames differ between assayData and phenoData

### DISCUSSION THREAD

**Discussion by sylestiel** [stuart-lab/signac#1544] [General]
Hi Tim,

 I was trying to generate a cds object with a view to getting the co-accessible sites. I'm getting the following error:

### int.cds <- as.CellDataSet(integrated,assay="ATAC", reduction = "wnn.umap")
Error in validObject(.Object) : 
  invalid class "CellDataSet" object: sampleNames differ between assayData and phenoData
In addition: Warning messages:
1: In .sparse2dense(e1) :
  sparse->dense coercion: allocating vector of size 24.5 GiB
2: In .sparse2dense(e2) :
  sparse->dense coercion: allocating vector of size 24.5 GiB

What follows are the specifics of my object:
> ### integrated
An object of class Seurat 
218108 features across 20983 samples within 4 assays 
Active assay: ATAC (156711 features, 156711 variable features)
 2 layers present: counts, data
 3 other assays present: RNA, SCT, integrated
### > integrated@reductions
$pca
A dimensional reduction object with key PC_ 
 Number of dimensions: 50 
 Number of cells: 20983 
 Projected dimensional reduction calculated:  FALSE 
 Jackstraw run: FALSE 
 Computed using assay: integrated 

$umap.rna
A dimensional reduction object with key rnaUMAP_ 
 Number of dimensions: 2 
 Number of cells: 20983 
 Projected dimensional reduction calculated:  FALSE 
 Jackstraw run: FALSE 
 Computed using assay: integrated 

$lsi
A dimensional reduction object with key LSI_ 
 Number of dimensions: 50 
 Number of cells: 20983 
 Projected dimensional reduction calculated:  FALSE 
 Jackstraw run: FALSE 
 Computed using assay: ATAC 

$umap.atac
A dimensional reduction object with key atacUMAP_ 
 Number of dimensions: 2 
 Number of cells: 20983 
 Projected dimensional reduction calculated:  FALSE 
 Jackstraw run: FALSE 
 Computed using assay: ATAC 

$wnn.umap
A dimensional reduction object with key wnnUMAP_ 
 Number of dimensions: 2 
 Number of cells: 20983 
 Projected dimensional reduction calculated:  FALSE 
 Jackstraw run: FALSE 
 Computed using assay: ATAC 

I integrated the RNA modality from 3 multiome assays but NOT the ATAC modalities since these were multiome datasets.. So there is overlap when I look at the wnn.umap projection and umap.rna projection but NOT with the umap.atac projection.
Is that to be expected?  Also to run LinkPeaks() I do have to establish the conns correct?
So which reduction should I be using? wnn.umap or umap.atac. 

What is the error:
invalid class "CellDataSet" object: sampleNames differ between assayData and phenoData referring to?  How may I fix it?

Despite adding the chr prefix to the ranges in the ATAC component of the object I'm still unable to run the Motif analysis. As a reminder I'm using the Macaca mulatta datasets and used the gtf file at Ensembl for Mmul_10 for annotation purposes.

Thanks for your time!

**Comment by timoast** [comment]
You need to use the `as.cell_data_set` function instead

> Also to run LinkPeaks() I do have to establish the conns correct?

No, if you want to run LinkPeaks() there's no need to run Cicero

**Reply by sylestiel** [reply]
Getting this error
integrated <- **LinkPeaks**(
+   object = integrated,
+   peak.assay = "ATAC",
+   expression.assay = "SCT",
+   genes.use = c("LEF1")
+ )
**Testing 1 genes and 156706 peaks
  |                                                  | 0 % ~calculating  Error in density.default(x = query.feature[[featmatch]], kernel = "gaussian",  : 
  argument 'x' must be numeric**


> # convert to CellDataSet format and make the cicero object
> int.cds <- as.cell_data_set(x = integrated)
Error in as.cell_data_set(x = integrated) : 
  could not find function "as.cell_data_set"
> # convert to CellDataSet format and make the cicero object
> int.cds <- as.CellDataSet(x = integrated)
Error in validObject(.Object) : 
  invalid class "CellDataSet" object: sampleNames differ between assayData and phenoData
In addition: Warning messages:
1: In .sparse2dense(e1) :
  sparse->dense coercion: allocating vector of size 24.5 GiB
2: In .sparse2dense(e2) :
  sparse->dense coercion: allocating vector of size 24.5 GiB

**Reply by sylestiel** [reply]
Hi@ Tim,

Still awaiting for a way to troubleshoot these problems.

Thanks,

**Reply by timoast** [reply]
Hi @sylestiel, please open an issue containing the full code you're running and the output of `sessionInfo()`

**Comment by sylestiel** [comment]
Full code as in from the very start of analysis?
